src/MainWindow.xaml.cs

```diff
@@ -27,5 +28,7 @@ private void MainWindow_Closed(object sender, WindowEventArgs args)
     {
         Application.Current.GetService<IExtensionService>().SignalStopExtensionsAsync();
         TelemetryFactory.Get<ITelemetry>().Log("DevHome_MainWindow_Closed_Event", LogLevel.Critical, new DevHomeClosedEvent(mainWindowCreated));
+        Log.Information("Terminating via MainWindow_Closed.");
+        Log.CloseAndFlush();

```

Comment:
From my one time trying it out, this prevents some other logging in shutdown. (Dashboard unloading happened after this). We should make sure there's nothing anywhere we need.

Reply:
I'll look into finding a better place for the log close and flush that is later in the shutdown path.

Reply:
Moved it to the end of Main, along with moving the logging to the beginning of main.